src/services/rebrandly/rebrandly.service.ts

```diff
@@ -0,0 +1,40 @@
+import { er } from "../services";
+import { EventsRepository } from "hackoss";
+import { PlatformService } from "../platform/platform.service";
+import * as Rebrandly from "rebrandly";
+import { RebrandlyLink } from "./link.rebrandly";
+import * as _ from 'lodash';
+export class RebrandlyService extends PlatformService{
+    private Rebrandly:any;
+    constructor(private eventsRepository:EventsRepository){
+        super();
+        
+        this.Rebrandly = new Rebrandly({
+            apiKey: process.env.REBRANDLY_API_KEY
+        });
+    }
+    async publish(eventId:string){
+        const event = await this.eventsRepository.getEvent(eventId);
+        await this.updateLink({ 
+            title: `Link for ${event.title}`, 
+            destination: event.githubUrl,
+            type: 'content'
+        });
+    }
+    async updateLink(link:RebrandlyLink){
+        //should not throw error, as it was not a failure
+        if(link.destination && link.destination.length !== 0){    
```

Comment:
Consider rewriting this as a guard clause. Eg.

```
if (something is wrong) {
    return;
}
```

Reply:
Then again, might be better to move validation into the routes file, not here.

Reply:
Routes would not know about events too, right? Maybe a better place is EventsRepository?

Reply:
I mean validation like missing params, or invalid params. This is up for discussion, but we can move all validation to the routes and then we just assume all is correct in the service. We can use the constraints defined here ntuoss/hackoss#7 as well.